### PR TITLE
Raw YAHPO Benchmarks

### PR DESCRIPTION
**Pull request by PhMueller**
Interface to evaluate the true performance of a configuration (not as surrogate or tabular benchmark)

### CODE REVIEW

File: extra_requirements/rbv2.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "rbv2": ["yahpo_gym@git+https://github.com/pfistfl/yahpo_gym#egg=yahpo_gym&subdirectory=yahpo_gym", "rpy2", "openml==0.10.2"],
+  "rbv2_old": ["yahpo_gym@git+https://github.com/pfistfl/yahpo_gym#egg=yahpo_gym&subdirectory=yahpo_gym", "rpy2", "numpy", "configspace", "pandas==1.2.4","scikit-learn==0.24.2", "openml==0.10.2"]
```
Comment (KEggensperger) [inline extra_requirements/rbv2.json]: there are missing dependency versions

File: hpobench/container/recipes/ml/Singularity.rbv2Benchmark [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    && Rscript -e 'remotes::install_version("mlr", version = "2.14", upgrade = "never")' \
+    && Rscript -e 'remotes::install_github("ml-org/mlr3misc", upgrade = "never")' \
+    && Rscript -e 'remotes::install_github("mlr-org/mlr3misc", upgrade = "never")'
+    && Rscript -e 'remotes::install_version("mlrCPO", version = "0.3.6", upgrade = "never")' \
+    && Rscript -e 'remotes::install_github("pfistlf/rbv2", upgrade = "never")'
+
+    cd /home \
+    && git clone https://github.com/pfistfl/HPOBench.git \
+    && cd HPOBench \
+    && echo "Please never push a recipe that checks out any other branch than development or master" \
```
Comment (KEggensperger) [inline hpobench/container/recipes/ml/Singularity.rbv2Benchmark]: This contradicts the next line where you checkout a different branch ...

**Review by KEggensperger — CHANGES_REQUESTED** [review]

File: hpobench/util/data_manager.py [hunk 3]
```diff
@@ -980,3 +981,36 @@ def _load_json(path):
...
+    def _try_download(self):
+        """Clone the data repository."""
+        if not self.data_dir.exists():
+            # Create the data directory if not existing
+            self.create_save_directory(self.data_dir.parent)
+
+            import git
+            git.Repo.clone_from(url='https://github.com/pfistfl/yahpo_data.git',
+                                to_path=str(self.data_dir), branch='main', multi_options=['--depth 1'])
+            self.logger.info(f'Successfully cloned data from repo to {self.data_dir}')
```
Comment (LGro) [inline hpobench/util/data_manager.py:1256]: Given that `yahpo_data` is licensed under [AGPL](https://github.com/slds-lmu/yahpo_data/blob/main/LICENSE), how do you judge the compatibility with this repo's more permissive Apache-2.0 license, @KEggensperger and @PhMueller?

Comment (LGro) [reply hpobench/util/data_manager.py:1256]: And the maintainers over there just [relicensed to Apache-2.0](https://github.com/slds-lmu/yahpo_data/commit/f9b8cd24fd8e7d9cc8fb7372f9640128941255e5). My question is resolved then, I think :relaxed:

**Pull request merged**